launch/launch/actions/execute_process.py

```diff
@@ -383,6 +406,8 @@ def _shutdown_process(self, context, *, send_sigint):
             # Do not handle shutdown more than once.
             return None
         self.__shutdown_received = True
```

Comment:
@ivanpauno 
I just wonder if the following code can be replaced?
https://github.com/ros2/launch/blob/f243fc6b3bcd276c194151e6933cda90761b1944/launch/launch/actions/execute_process.py#L739-L741

Because **self.__shutdown_future = create_future(context.asyncio_loop)** is created after it.

Reply:
Good catch, that check doesn't have any sense :joy:.
The shutdown event handler is installed **after** that check, so `self.__shutdown_received` will always be true in that place.

That should be replaced with `context.is_shutdown()`.